### PR TITLE
Fix API endpoints for related media and provider stats

### PR DESCRIPTION
**Pull request by zackkrida**
## Fixes

This PR fixes the endpoints to fetch related media, updates to use the new endpoints for provider stats, and removes fallback default provider data that was hardcoded into the repo, which can give false assurances that things are working. 

We'll need to test API changes on the front-end much more closely moving forward. 

Noting that in production, the related API endpoint is still seemingly broken, although my local instance of the API is working. 

## Description

Some additional low-hanging cleanup here:

- Fix audio endpoints from 'audios' to 'audio'
- Remove unused provider collection endpoints
- Add some jsdoc annotations for axios
- Fix error handling of `getProviderStats`, which wouldn't actually catch exceptions
- Start the work to make the Sources page media-type agnostic (it still doesn't actually fetch the audio providers)

## Testing Instructions

- Run locally and observe the provider list on the 'sources' page and in the results filter sidebar is visible.
- Run tests and ensure they work
- Bonus: Run a local instance of the API and change the nuxt.config.js to point to it (don't forget `v1` at the end of the endpoint!). Confirm that locally configured sources are visible.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Review by sarayourfriend — APPROVED** [review]
LGTM, tests great. Hopefully we can figure out why related images are broken but at least the critical path is fixed.

**Review by AetherUnbound — APPROVED** [review]
Looks good!

File: src/data/api-service.js [hunk 1]
```diff
@@ -23,26 +23,59 @@ export const createApiService = (baseUrl = process.env.apiUrl) => {
...
       return client.put(`${resource}/${slug}`, params, { headers })
     },
 
+    /**
+     * @param {string} resource  The endpoint of the resource
+     * @param {import('axios').AxiosRequestConfig} params Url parameter object
+     * @returns {Promise<import('axios').AxiosResponse<any>>} Response The API response object
+     */
     put(resource, params) {
       return client.put(`${resource}`, params)
```
Comment (obulat) [inline src/data/api-service.js:70]:
```suggestion
      return client.put(resource, params)
```

File: src/store/provider.js [hunk 2]
```diff
@@ -57,13 +50,13 @@ export const createActions = (services) => ({
     return providerService
       .getProviderStats()
       .then(({ data }) => {
+        console.log(data)
         sortedProviders = sortProviders(data)
```
Comment (obulat) [inline src/store/provider.js:54]:
```suggestion
        sortedProviders = sortProviders(data)
```

**Review by obulat — APPROVED** [review]
I tried it with the local API, and everything works great!
I added code suggestions to remove a stray log statement and a small change.

**Comment by obulat** [comment]
I get the 400 error for related endpoint, it is caused by pagination error in the API:
https://github.com/WordPress/openverse-api/blob/13673ee8c196c9a6b786a1322d6d65b97c6ca708/openverse_api/catalog/api/utils/pagination.py#L29

**Pull request merged**